### PR TITLE
spike post purchase survey solution with tally + CIO

### PR DESCRIPTION
**Pull request by zacjones93**
I've spiked out a solution for the post-purchase survey. 

Currently, the component is living on the `/post-purchase` page with the proper context to display the page pulled in so that I didn't have to have a proper session or deal with stripe. i.e. `session` is set to `{email: 'zac@egghead.io', amount: '25.00'}` and `currentState` set to `'authTokenRetrieved'` so that the confirmation page would present as if a transaction was successful. 

## Tally + Customer.io
The post purchase survey is created in tally.so. 

When an entry is completed in Tally, that entry is uploaded to google sheets. 

There is a Zapier integration watching the sheet for new row entries. It will take the email from the row entry and assign a `post_purchase_survey` attribute to a customer in CIO matching the email so we know they've completed the survey.

<details>
<summary>Screenshots</summary>

<br>

Build the portfolio you need to be a badass web developer   egghe
Build the portfolio you need to be a badass web developer   egghe

</details>

### CODE REVIEW

File: src/pages/post-purchase.tsx [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+        button="Submit Feedback"
+        onSubmit={onSubmit}
+      >
+        <div className="text-center">
+          <p className="mt-4 text-base font-normal text-black dark:text-white sm:text-lg">
+            {question}
+          </p>
+        </div>
+      </Form>
+    </div>
```
Comment (zacjones93) [inline src/pages/post-purchase.tsx:56]: This is essentially the `OnlinePresenceEmailEntryForm` repurposed for this context

File: src/pages/post-purchase.tsx [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+
+  let onSubmit = async (values: any) => {
+    const {response} = values
+
+    await supabase.from('responses').insert({
+      email: viewer?.email,
+      question: question,
+      response: response,
+    })
+  }
```
Comment (zacjones93) [inline src/pages/post-purchase.tsx:40]: supabase is easy to work with

File: src/pages/post-purchase.tsx [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+  HeaderImageComponent?: React.FC
+  className?: string
+  button?: string
+  label?: string
+  formClassName?: string
+  onSubmit: (value: any, setIsError: any) => void
+}
+
+const PostPurchaseSurvey: FunctionComponent<any> = ({
+  question = 'What was the last thing that held you back from purchasing?',
```
Comment (zacjones93) [inline src/pages/post-purchase.tsx:28]: we could change the question whenever

File: src/pages/post-purchase.tsx [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+              </p>
+            </div>
+          )}
+        </div>
+      </div>
+    </div>
+  )
+}
+
+//! delete
```
Comment (zacjones93) [inline src/pages/post-purchase.tsx:197]: If we were to use this, I would delete from here on and move `PostPurchaseSurvey` into the components folder and use in `NewMemberConfirmation`

File: src/pages/post-purchase.tsx [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+              <PopularTopics />
+            </div>
+            <div className="flex justify-center pt-6">
+              <StartLearning />
+            </div>
+          </>
+        }
+      />
+      <Support />
+    </>
```
Comment (zacjones93) [inline src/pages/post-purchase.tsx:250]: Survey is above the fold

**Review by joelhooks — APPROVED** [review]

**Comment by joelhooks** [comment]
accidentally approved 😭

File: src/components/pages/confirm/membership/index.tsx [hunk 6]
```diff
@@ -207,6 +209,12 @@ const NewMemberConfirmation: React.FC<{session: any; currentState: any}> = ({
   session,
   currentState,
 }) => {
+  const {_, setTheme} = useTheme()
+
+  React.useEffect(() => {
+    setTheme('dark')
+  }, [setTheme])
+
```
Comment (zacjones93) [inline src/components/pages/confirm/membership/index.tsx:217]: sets theme to dark so survey can be styled consistently

File: src/components/pages/confirm/membership/index.tsx [hunk 7]
```diff
@@ -241,20 +249,23 @@ const NewMemberConfirmation: React.FC<{session: any; currentState: any}> = ({
...
+                <p className="max-w-lg pb-8 mx-auto text-lg text-center border-b border-gray-100">
                   We've charged your credit card{' '}
                   <strong>${session.amount} for an egghead membership</strong>{' '}
                   and sent a receipt to <strong>{session.email}</strong>. Please
                   check your inbox to{' '}
                   <strong>confirm your email address</strong>.
                 </p>
+
+                <PostPurchase email={session?.email} />
+
```
Comment (zacjones93) [inline src/components/pages/confirm/membership/index.tsx:268]: Pass the session email to `PostPurchase` survey

File: src/components/survey/tally/post-purchase.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+        frameBorder={1}
+        marginHeight={1}
+        marginWidth={1}
+        title="Post Purchase Question"
+      ></iframe>
+    </div>
+  )
+}
+
+export default PostPurchase
```
Comment (zacjones93) [inline src/components/survey/tally/post-purchase.tsx:17]: I have the form hosted on my domain from my tally account but can switch this over. On the tally side of things I'm overriding styles
```
<style>
h3 {
  color: #ffffff !important;
}
button {
  background-color: #2f6af4 !important;
}
</style>
```

**Review by zacjones93 — COMMENTED** [review]
I haven't been able to test Stripe Checkout locally so would like a sanity check there before merging but this works on `/post-purchase`

**Comment by zacjones93** [comment]
Build the portfolio you need to be a badass web developer   egghe

**Comment by zacjones93** [comment]
Build the portfolio you need to be a badass web developer   egghe

**Review by Creeland — APPROVED** [review]
Added integromat integration 
image

**Pull request merged**